public/src/components/channelManagement/bannerTests/bannerTestEditor.tsx

```diff
@@ -114,20 +110,39 @@ const BannerTestEditor: React.FC<BannerTestEditorProps> = ({
     return undefined;
   };
 
-  const updateTest = (updatedTest: BannerTest): void => {
-    onChange({
-      ...updatedTest,
-      // To save dotcom from having to work this out
-      articlesViewedSettings: getArticlesViewedSettings(updatedTest),
-    });
+  const updateTest = (update: (test: BannerTest) => BannerTest): void => {
+    if (test) {
+      const updatedTest = update(test);
+
+      onChange({
+        ...updatedTest,
+        // To save dotcom from having to work this out
+        articlesViewedSettings: getArticlesViewedSettings(updatedTest),
+      });
+    }
```

Comment:
How come we're making this change?

Reply:
Basically to better mirror the implementation on epics. My thinking is that this will help pave the way to share as much code as possible between the two.

Reply:
Yeah that makes sense - I think it might be best to refactor the other way around. The epic manager still has some older logic from the old style UX of the tool e.g the `if (test)` isn't really needed as the type of `test` is `BannerTest` so it can't be `null/undefined`.